subsys/bootloader/cmake/sign.cmake

```diff
@@ -67,32 +81,71 @@ endif()
 
 add_custom_command(
   OUTPUT
-  ${signature_file}
+  ${HASH_FILE}
   COMMAND
-  ${hashcmd}
+  ${HASH_CMD}
+  DEPENDS
+  ${KERNEL_ELF_TO_BE_SIGNED}
+  WORKING_DIRECTORY
+  ${PROJECT_BINARY_DIR}
+  COMMENT
+  "Creating hash of application"
+  USES_TERMINAL
+  )
+
+add_custom_target(
+  hash_file_target
+  DEPENDS
+  ${HASH_FILE}
+  ${KERNEL_ELF_TO_BE_SIGNED}
+  generated_file_directory_target
+  )
+
+add_custom_command(
+  OUTPUT
+  ${SIGNATURE_FILE}
   COMMAND
-  ${signcmd}
+  ${SIGN_CMD}
   DEPENDS
-  ${sign_depends}
-  WORKING_DIRECTORY ${PROJECT_BINARY_DIR}
+  ${KERNEL_ELF_TO_BE_SIGNED}
+  ${HASH_FILE}
```

Comment:
Could you depend on `hash_file_target` instead? That should be the only `DEPENDS` item needed in that case.

Also, could these two commands be merged in to a single `add_custom_command` which contains two `COMMAND` arguments? The `COMMAND` arguments are executed in the order they are specified. If that is possible you will get rid of some overhead with creating targets for everything.

Reply:
could also remove `kernel_elf` I'll merge them back into one.